lotto/offchain/src/MaybeMalformed.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | A module to easily inject malformed pieces of data in well-typed haskell
+-- code. A 'MaybeMalformed' piece of data can be either well-formed, in which
+-- case it carries a value of the expected type, or malformed, in which case it
+-- carries an arbitrary Plutus builtin data.
+module MaybeMalformed
+  ( MaybeMalformed (..),
+    fromWellFormed,
+    fromMalformed,
+    wellFormed,
+    malformed,
```

Comment:
Yes, I also saw that while writing and was a bit puzzled. On one hand `wellFormed` is a better camel case for the words “well-formed” and people would expect that, but on the other hand there is a weird discrepancy between `wellFormed` and `malformed` and that might be error prone. I can't decide what to do.

Also not that I'm very much not attached to the wording, it's just the only one I could find (and I tried to avoid words such as “normal” that are way overloaded). Also, it's a bit weird because well-formedness is more a property that applies to the serialised version of the data _with respect to_ a Haskell type, but here the constructors `WellFormed` and `Malformed` contain properties on the Haskell level, so I'm not sure what to make of that either.